govcd/vm.go

```diff
@@ -252,22 +253,19 @@ func (vm *VM) ChangeNetworkConfig(networks []map[string]interface{}, ip string)
 				ipAddress := "Any"
 
 				// TODO: Review current behaviour of using DHCP when left blank
-				if ip == "dhcp" || network["ip"].(string) == "dhcp" {
+				if network["ip"].(string) == "dhcp" {
```

Comment:
This is pretty confusing to compute on the brain.
I've tried to automated input/output testing for this piece of code and it seems to me that the output's are different which may break someone's (not necessarily terraform provider) functionality.
Here's the sample code: https://gist.github.com/Didainius/6c8b8cdc75d57d3af9bc16190b742c7b
And it's output in the bottom (you can notice if `ip` is empty you get different results).

Can you verify if it's my mistake or difference in outputs.

Reply:
I guess you did not supply somthing for network["ip_allocation_mode"].(string) when you left ip 
 and network["ip"].(string) empty right?

Reply:
Well to have the exact same behaviour as before I would need to check if network["ip_allocation_mode"].(string) is empty and if it is apply dhcp otherwise apply network["ip_allocation_mode"].(string).

Reply:
Looks like you're right. I'd like that function to still be 100% compatible with previous. Looking at codebase for terraform it didn't set `network["ip_allocation_mode"].(string)` before triggering this function so while I'm pretty fresh to this codebase myself, I believe it may break things for someone. Another way could be what vbauzysvmware mentioned below - introduce a new function and wrap it in this one for backwards compatibility.

Reply:
Behavior is fixed.